src/components/DoctorDataTable.vue

```diff
@@ -0,0 +1,272 @@
+import { ref, onMounted } from 'vue';
+import { FilterMatchMode } from 'primevue/api';
+import { useToast } from 'primevue/usetoast';
+import { getDoctors, deleteDoctor, createDoctor, editDoctor } from '@/services/api'
+
+const toast = useToast();
+const loading = ref(true)
+const dt = ref();
+const doctors = ref();
+const doctorDialog = ref(false);
+const editingDoctor = ref(false);
+const deleteDoctorDialog = ref(false);
+const deleteDoctorsDialog = ref(false);
+const doctor = ref({});
+const selectedDoctors = ref();
+const filters = ref({
+    'global': {value: null, matchMode: FilterMatchMode.CONTAINS},
+});
+const submitted = ref(false);
+const pagination = ref({
+    currentPage: 0,
+    totalPages: 0,
+    currentRecords: 0,
+    totalRecords: 0
+})
+
+const loadDoctors = (page = 0) => {
+    loading.value = true;
+    getDoctors(page)
+        .then(response => {
+            pagination.value = response.pagination
+            doctors.value = response.data
+        })
+        .catch(err => {
+            console.error(err);
+            toast.add({severity:'error', summary: err?.response?.statusText || 'Error', detail: err?.response?.data?.msg || 'Could not fetch doctors', life: 3000});
+        })
+        .finally(() => loading.value = false)
+}
+
+onMounted(() => {
+    loadDoctors();
+})
+
+const openNew = () => {
+    doctor.value = {};
+    submitted.value = false;
+    doctorDialog.value = true;
+};
+const hideDialog = () => {
+    doctorDialog.value = false;
+    submitted.value = false;
+};
+const saveNewDoctor = () => {
+    submitted.value = true;
+    if(!doctor.value.firstName || !doctor.value.lastName || !doctor.value.email || !doctor.value.password) return;
+    createDoctor(doctor.value.firstName, doctor.value.lastName, doctor.value.email, doctor.value.password)
+        .then(() => {
+            toast.add({severity:'success', summary: 'Success', detail: 'Doctor information saved', life: 3000});
+            loadDoctors();
+        })
+        .catch(err => {
+            console.error(err);
+            toast.add({severity:'error', summary: err?.response?.statusText || 'Error', detail: err?.response?.data?.msg || 'Could not create doctor', life: 3000});
+        })
+        .finally(() => {
+            doctorDialog.value = false;
+            doctor.value = {};
+        })
+};
+
+const saveEditedDoctor = () => {
+    submitted.value = true;
+    if(!doctor.value.firstName || !doctor.value.lastName || !doctor.value.email) return;
+    editDoctor(doctor.value.id, doctor.value.firstName, doctor.value.lastName, doctor.value.email)
+        .then(() => {
+            toast.add({severity:'success', summary: 'Success', detail: 'Doctor information saved', life: 3000});
+            loadDoctors();
+        })
+        .catch(err => {
+            console.error(err);
+           toast.add({severity:'error', summary: err?.response?.statusText || 'Error', detail: err?.response?.data?.msg || 'Could not edit doctor', life: 3000});
```

Comment:
Może zrobić jakiegoś helpoera żeby te errory ogarniał? Bo teraz w każdym miejscu jest taki szlaczek

Reply:
Tak, to już zaplanowałem do zmiany przy porządkach - task jest na jirze